dts/bindings/w1/adi,max32-w1.yaml

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) 2023-2024 Analog Devices, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+description: ADI MAX32xxx MCUs 1-Wire Master
+
+include: [w1-master.yaml, pinctrl-device.yaml]
+
+compatible: "adi,max32-w1"
+
+properties:
+  reg:
+    required: true
+
+  clocks:
+    required: true
+
+  interrupts:
+    required: true
+
+  pinctrl-0:
+    required: true
+
+  pinctrl-names:
+    required: true
+
+  internal-pullup:
+    required: true
+    type: int
+    enum: [0, 1]
+    description: |
+      Set this field to enable the internal pullup resistor.
+      0 - Internal pullup disabled.
+      1 - Internal pullup enabled.
+
+  external-pullup:
+    required: true
+    type: int
+    enum: [0, 1, 2]
+    description: |
+      Set this field to enable the external pullup.
+      0 - Pullup pin is active high when enabled.
+      1 - Pullup pin is active low when enabled.
+      2 - Pullup pin is not used for an external pullup.
+
+  long-line-mode:
+    required: true
+    type: int
+    enum: [0, 1]
```

Comment:
suggestion is to use string enum but not required

Reply:
Thanks for the feedback, I tried to match it with UserGuide, it might be more clear if we keep it same as UG.
![image](https://github.com/zephyrproject-rtos/zephyr/assets/46590392/6c3b7b62-10e7-49a9-9f0b-14f7350bd7a4)

Reply:
Update: Converted to boolean as per of @str4t0m feedback. Please see: https://github.com/zephyrproject-rtos/zephyr/pull/75009#discussion_r1730874035